nautobot_secrets_providers/providers/thycotic.py

```diff
@@ -0,0 +1,247 @@
+            #   def    |    -     |  def  |   *    | AccessTokenAuthorizer
+            #    -     |   def    |  def  |   *    | AccessTokenAuthorizer
+            if username is not None and password is not None:
+                if domain is not None:
+                    thy_authorizer = DomainPasswordGrantAuthorizer(
+                        base_url=base_url,
+                        domain=domain,
+                        username=username,
+                        password=password,
+                    )
+                else:
+                    thy_authorizer = PasswordGrantAuthorizer(
+                        base_url=base_url,
+                        username=username,
+                        password=password,
+                    )
+            else:
+                thy_authorizer = AccessTokenAuthorizer(token)
+
+            # Get the client.
+            if cloud_based:
+                thycotic = SecretServerCloud(tenant=tenant, authorizer=thy_authorizer)
+            else:
+                thycotic = SecretServer(base_url=base_url, authorizer=thy_authorizer)
+
+            # Attempt to retrieve the secret.
+            try:
+                if secret_id is not None:
+                    secret = ServerSecret(**thycotic.get_secret(secret_id))
+                else:
+                    secret = ServerSecret(**thycotic.get_secret_by_path(secret_path))
+            except SecretServerError as err:
+                raise exceptions.SecretValueNotFoundError(secret, caller_class, str(err)) from err
+
+            # Attempt to return the selected value.
+            try:
+                return secret.fields[secret_selected_value].value
+            except KeyError as err:
+                msg = f"The secret value could not be retrieved using key {err}"
+                raise exceptions.SecretValueNotFoundError(secret, caller_class, msg) from err
+        finally:
+            if must_restore_env:
+                os.environ["REQUESTS_CA_BUNDLE"] = original_env
+
+
+class ThycoticSecretServerSecretsProviderId(ThycoticSecretServerSecretsProviderBase):
+    """A secrets provider for Thycotic Secret Server."""
+
+    slug = "thycotic-tss-id"  # type: ignore
+    name = "Thycotic Secret Server by ID"  # type: ignore
+
+    class ParametersForm(BootstrapMixin, forms.Form):
+        """Required parameters for Thycotic Secret Server."""
+
+        secret_id = forms.IntegerField(
+            label="Secret ID",
+            required=True,
+            min_value=1,
+            help_text="The secret-id used to select the entry in Thycotic Secret Server.",
+        )
+        secret_selected_value = forms.ChoiceField(
+            label="Return value",
+            required=True,
+            choices=(
+                ("password", "Password"),
+                ("username", "Username"),
+                ("url", "URL"),
+                ("notes", "Notes"),
+            ),
```

Comment:
Since you're using this same choice set in both providers, can you please make this into a `ChoiceSet` defined at the module-level and passed to the `ChoiceField.choices` for each?

```python
from nautobot.utilities.choices import ChoiceSet

class ThycoticSecretChoices(ChoiceSet):
    SECRET_PASSWORD = "password"
    SECRET_USERNAME = "username"
    SECRET_URL = "url"
    SECRET_NOTES = "notes"

    CHOICES = (
        (SECRET_PASSWORD, "Password"),
        (SECRET_USERNAME, "Username"),
        (SECRET_URL, "URL"),
        (SECRET_NOTES, "Notes"),
    )

# And then on the field for each provider's `ParametersForm`:
            choices=ThycoticSecretChoices,